lib/internal/bootstrap/pre_execution.js

```diff
@@ -136,6 +137,51 @@ function initializeDeprecations() {
                 'DEP0103') :
       types[name];
   }
+
+  const { cjsContext } = NativeModule;
+  const { getStackSourceName } = internalBinding('contextify');
+  const experimentalModules = getOptionValue('--experimental-modules');
+
+  // Deprecate global.process and global.Buffer access in ES Modules.
+  const processGetter = deprecate(
+    () => cjsContext.process,
+    'global.process is deprecated in ECMAScript Modules. ' +
+        'Please use `import process from \'process\'` instead.');
```

Comment:
Please add a deprecation code. This is done for all deprecations so far.

Reply:
@joyeecheung suggested above:

> I believe we do not have to go through normal deprecation process if this is limited to ESM, which is experimental anyways - it needed a doc-deprecation previously because it affected both ESM and CJS.

I removed the deprecation code and documentation based on this feedback, but have added back the deprecation codes for now, just without documenting them as a middle-ground approach.